### PR TITLE
Make locale setting also change thread culture

### PR DESCRIPTION
**Pull request by jorolf**
Fixes https://github.com/ppy/osu/issues/6409

Humanizer uses the current thread culture to humanize numbers/dates. This should make it use the correct culture.

### CODE REVIEW

File: osu.Framework/Platform/GameHost.cs [hunk 1]
```diff
@@ -784,6 +787,21 @@ protected virtual void SetupConfig(IDictionary<FrameworkSetting, object> gameDef
             }, true);
 
             bypassFrontToBackPass = DebugConfig.GetBindable<bool>(DebugSetting.BypassFrontToBackPass);
+
+            threadLocale = Config.GetBindable<string>(FrameworkSetting.Locale);
+            threadLocale.BindValueChanged(locale =>
+            {
+                var culture = string.IsNullOrWhiteSpace(locale.NewValue) ? CultureInfo.InvariantCulture : CultureInfo.GetCultures(CultureTypes.AllCultures).FirstOrDefault(c => c.Name.Equals(locale.NewValue, StringComparison.InvariantCultureIgnoreCase)) ?? CultureInfo.InvariantCulture;
```
Comment (bdach) [inline osu.Framework/Platform/GameHost.cs]: I might be missing something here since this line is really long and hard to parse, but couldn't you just do
```csharp
var culture = Culture.GetCultureInfo(locale.NewValue ?? string.Empty);
```
here? `Culture.GetCultureInfo` is [documented to search by name without case sensitivity](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.getcultureinfo?view=netcore-3.0#System_Globalization_CultureInfo_GetCultureInfo_System_String_), and the invariant culture [is documented to have an empty name](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.0).

Comment (jorolf) [reply osu.Framework/Platform/GameHost.cs]: Not sure if we want to use that method though:
>  if the operating system does not support that culture, [...], the method throws a CultureNotFoundException.

Comment (jai-x) [reply osu.Framework/Platform/GameHost.cs]: I think it would still be good to break up this line to make it more readable, even if we use the same method.

Comment (bdach) [reply osu.Framework/Platform/GameHost.cs]: Depends on what semantics we're after here, I suppose. At the framework level I'd probably rather have an overt exception instead of a silent fallback.

Comment (jorolf) [reply osu.Framework/Platform/GameHost.cs]: Note that the exception instantly crashes the framework on startup. If the locale setting somehow got corrupted it will stop the framework from starting.

**Comment by smoogipoo** [comment]
@peppy requesting review since I've made a few changes here.
